### DISCUSSION TITLE
Dmx rdm

### DISCUSSION THREAD

**Discussion by surajchauhan6631** [someweisguy/esp_dmx#131] [General]
I want to make a 3 channel dmx receiver based on esp32. And for setting the address of this receiver I want to use rdm functionality of your library. But I am confused where to start for using this rdm functionality. So help me how can I use this dmx rdm library for base address setting

**Comment by someweisguy** [comment]
Thanks for reaching out. So that I may better assist you, can you clarify which parts of the documentation you are having difficulties understanding?

**Reply by surajchauhan6631** [reply]
Your library is well documented, but I am little confused about how to use rdm functionality to set dmx device base address, can you share an example where you can show how we can set dmx address using rdm , that will help a lot

**Reply by someweisguy** [reply]
Absolutely - take a look at the [Arduino_RDMController](https://github.com/someweisguy/esp_dmx/blob/release/v4.0/examples/Arduino_RDMController/Arduino_RDMController.ino#L135-L158) example. This shows an example of the `rdm_send_set_dmx_start_address()` function. 

The gist is that in order to find devices on the RDM network, you typically will need to run the RDM discovery algorithm. This library provides the function `rdm_discover_devices_simple()` to find RDM devices on the network. This function gets an array of RDM device UIDs on the network. Once you have a device UID, you can use it to send RDM requests such as `rdm_send_set_dmx_start_address()`.

The example code linked above performs RDM discovery and prints device info for each discovered device. It also swaps the RDM identify state and increments the DMX start address of each discovered device by one. If you prefer ESP-IDF, there is an ESP-IDF version of the linked example code [here](https://github.com/someweisguy/esp_dmx/blob/release/v4.0/examples/ESPIDF_RDMController/ESPIDF_RDMController.c).

Hopefully this helps!
